src/search/searcher.h

```diff
@@ -459,7 +461,14 @@ class Searcher : public std::enable_shared_from_this<Searcher> {
             }
         }
 
-        core::TranspositionTable::writeEntry(m_stackItr->board.hash, bestScore, m_stackItr->eval, bestMove, ttPv, depth, m_ply, ttFlag);
+        if (!isChecked
+            && bestMove.isQuietMove()
+            && !(ttFlag == core::TtFlag::TtAlpha && bestScore >= m_stackItr->eval)
```

Comment:
Could you please explain these conditions?

Reply:
It's the basic TT cutoff conditions. We're doing the same whenever checking the bounds :)

Reply:
Added comments about it just to clarify :)